packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -210,14 +215,28 @@ class WebFs {
     final HttpServer server = await httpMultiServerFactory(_kHostName, port);
     shelf_io.serveRequests(server, cascade.handler);
     final Chrome chrome = await chromeLauncher.launch('http://$_kHostName:$port/');
+    final WipConnection wipConnection = await _connectToTab(chrome);
     return WebFs(
       client,
       server,
       dwds,
       chrome,
+      wipConnection,
     );
   }
 
+  static Future<WipConnection> _connectToTab(Chrome chrome) async {
+    final List<ChromeTab> tabs = await chrome.chromeConnection.getTabs();
+    for (ChromeTab tab in tabs) {
+      if (!tab.url.contains('localhost')) {
```

Comment:
I know it isn't a new issue from this CL but checking for chrome tabs on localhost doesn't seem robust. For example, dwds does support launching on a different hostname, there could be other tabs open matching the hostname, etc. Internally, dwds has support to figure out exactly which tab it cares about by checking the content of the tab for the magic ids. Is there a reason you can't delegate this task to dwds.